api/src/main/java/com/okta/sdk/lang/Locales.java

```diff
@@ -0,0 +1,202 @@
+     */
+    private static Locale parseLocale(final String str) {
+        if (isISO639LanguageCode(str)) {
+            return new Locale(str);
+        }
+
+        final String[] segments = str.split("_", -1);
+        final String language = segments[0];
+        if (segments.length == 2) {
+            final String country = segments[1];
+            if (isISO639LanguageCode(language) && isISO3166CountryCode(country) ||
+                    isNumericAreaCode(country)) {
+                return new Locale(language, country);
+            }
+        } else if (segments.length == 3) {
+            final String country = segments[1];
+            final String variant = segments[2];
+            if (isISO639LanguageCode(language) &&
+                    (country.length() == 0 || isISO3166CountryCode(country) || isNumericAreaCode(country)) &&
+                    variant.length() > 0) {
+                return new Locale(language, country, variant);
+            }
+        }
+        throw new IllegalArgumentException("Invalid locale format: " + str);
+    }
+
+     /**
+     * Checks whether the given String is a ISO 639 compliant language code.
+     *
+     * @param str the String to check.
+     * @return true, if the given String is a ISO 639 compliant language code.
+     */
+    private static boolean isISO639LanguageCode(final String str) {
+        return isAllLowerCase(str) && (str.length() == 2 || str.length() == 3);
+    }
+
+    /**
+     * Checks whether the given String is a ISO 3166 alpha-2 country code.
+     *
+     * @param str the String to check
+     * @return true, is the given String is a ISO 3166 compliant country code.
+     */
+    private static boolean isISO3166CountryCode(final String str) {
+        return isAllUpperCase(str) && str.length() == 2;
+    }
+
+    /**
+     * Checks whether the given String is a UN M.49 numeric area code.
+     *
+     * @param str the String to check
+     * @return true, is the given String is a UN M.49 numeric area code.
+     */
+    private static boolean isNumericAreaCode(final String str) {
+        return isNumeric(str) && str.length() == 3;
+    }
+
+    private static boolean isAllLowerCase(final CharSequence cs) {
+        if (cs == null || Strings.isEmpty(cs)) {
+            return false;
+        }
+        final int sz = cs.length();
+        for (int i = 0; i < sz; i++) {
+            if (!Character.isLowerCase(cs.charAt(i))) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private static boolean isAllUpperCase(final CharSequence cs) {
+        if (cs == null || Strings.isEmpty(cs)) {
+            return false;
+        }
+        final int sz = cs.length();
+        for (int i = 0; i < sz; i++) {
+            if (!Character.isUpperCase(cs.charAt(i))) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private static boolean isNumeric(final CharSequence cs) {
```

Comment:
It might make more sense to pull these methods out into a generic Util file and live under `/imp/util`.

Reply:
Not a bad idea, but I don't think they would be used outside this context (and i'd rather keep them private until needed)